subsys/ipc/ipc_service/lib/Kconfig.icmsg

```diff
@@ -27,15 +27,53 @@ config IPC_SERVICE_ICMSG_SHMEM_ACCESS_TO_MS
 	  Maximum time to wait, in milliseconds, for access to send data with
 	  backends basing on icmsg library. This time should be relatively low.
 
-# The Icmsg library in its simplicity requires the system workqueue to execute
-# at a cooperative priority.
-config SYSTEM_WORKQUEUE_PRIORITY
-	range -256 -1
-
 config IPC_SERVICE_ICMSG_BOND_NOTIFY_REPEAT_TO_MS
 	int "Bond notification timeout in miliseconds"
 	range 1 100
 	default 1
 	help
 	  Time to wait for remote bonding notification before the
 	  notification is repeated.
+
+config IPC_SERVICE_BACKED_ICMSQ_WQ_ENABLE
+	bool "Use dedicated workqueue"
+	default y
```

Comment:
Can we make it disabled by default? The issue is we are not sure if everything in the system is compatible with this configuration. I am not saying it is not - but we did not check that yet.

Reply:
Enabling this by default makes the backend compatible in usage scenarios where RPMsg has been currently used. From that point of view, this approach follows the principle of least surprise.